lib/IDE/CursorInfo.cpp

```diff
@@ -175,7 +175,12 @@ class NodeFinder : ASTWalker {
   }
 
   PreWalkAction walkToDeclPre(Decl *D) override {
-    if (!rangeContainsLocToResolve(D->getSourceRangeIncludingAttrs())) {
+    // If the decl doesn't contain the location to resolve, we can skip walking
+    // it. One exception to this is for VarDecls, they can contain accessors
+    // which are not included in their SourceRange. For e.g `var x: Int { 0 }`,
+    // the VarDecl's range is just `x`, but the location may be in the accessor.
+    if (!isa<VarDecl>(D) &&
+        !rangeContainsLocToResolve(D->getSourceRangeIncludingAttrs())) {
       return Action::SkipNode();
```

Comment:
Is it worth only skipping those with accessors, or is that too hard/expensive to check?

Reply:
We only walk accessors for VarDecls, so if there are no accessors this would basically be a no-op anyways:

https://github.com/swiftlang/swift/blob/588a7db17cda0b0595f87bbfc97a2af1369d8c08/lib/AST/ASTWalker.cpp#L351-L359

Reply:
Ah, fair enough 👍